state-chain/pallets/cf-elections/src/electoral_systems/tests/delta_based_ingress.rs

```diff
@@ -0,0 +1,383 @@
+	BTreeMap::from_iter(
+		state_map.into_iter().map(|(key, value)| (key.encode(), Some(value.encode()))),
+	)
+}
+
+fn initial_channel_state() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 0u64,
+			block_number: 0u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 0u64,
+			block_number: 0u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+fn channel_state_ingressed() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 1_000u64,
+			block_number: 700u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 2_000u64,
+			block_number: 800u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn channel_state_reverted() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 500u64,
+			block_number: 900u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 1_000u64,
+			block_number: 900u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn channel_state_final() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 2_000u64,
+			block_number: 900u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 4_000u64,
+			block_number: 900u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn channel_state_closed() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 2_000u64,
+			block_number: 1_000u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 4_000u64,
+			block_number: 2_000u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn with_setup(initial_elections: Option<DepositChannels>) -> TestSetup<SimpleDeltaBasedIngress> {
+	let initial_elections = initial_elections.unwrap_or(initial_channel_state());
+	TestSetup::<_>::default()
+		.with_initial_election_state(
+			1u32,
+			initial_elections.to_properties(),
+			initial_elections.to_state(),
+		)
+		.with_initial_state_map(initial_elections.to_state_map().into_iter().collect::<Vec<_>>())
+}
+
+register_checks! {
+	SimpleDeltaBasedIngress {
+		started_at_default_state(pre_finalize, _post) {
+			assert_eq!(
+				pre_finalize.unsynchronised_state_map,
+				to_raw_state_map(initial_channel_state().to_state_map()),
+				"Expected initial state incorrect pre-finalization."
+			);
+		},
+		ended_at_ingressed_state(_pre, post_finalize) {
+			assert_eq!(
+				post_finalize.unsynchronised_state_map,
+				to_raw_state_map(channel_state_ingressed().to_state_map()),
```

Comment:
we could also probably use the parameterised checks here too, since we're creating local state in the test and comparing to a global const, we can pass through the local expected state as a param inside the test so it's all nice and contained